doc/man7/migration_guide.pod

```diff
@@ -0,0 +1,1807 @@
+
+=item L<X509_NAME_hash(3)>
+
+=item L<X509_new(3)>
+
+=item L<X509_REQ_new(3)> and L<X509_REQ_verify(3)>
+
+=item L<X509_STORE_CTX_new(3)>, L<X509_STORE_set_default_paths(3)>, L<X509_STORE_load_file(3)>,
+L<X509_STORE_load_locations(3)> and L<X509_STORE_load_store(3)>
+
+=back
+
+=head4 New functions that use a Library context
+
+The following functions can be passed a library context if required.
+Passing NULL will use the default library context.
+
+=over 4
+
+=item L<EVP_ASYM_CIPHER_fetch(3)> and L<EVP_ASYM_CIPHER_do_all_provided(3)>
+
+=item L<EVP_CIPHER_fetch(3)> and L<EVP_CIPHER_do_all_provided(3)>
+
+=item L<EVP_default_properties_enable_fips(3)> and
+L<EVP_default_properties_is_fips_enabled(3)>
+
+=item L<EVP_KDF_fetch(3)> and L<EVP_KDF_do_all_provided(3)>
+
+=item L<EVP_KEM_fetch(3)> and L<EVP_KEM_do_all_provided(3)>
+
+=item L<EVP_KEYEXCH_fetch(3)> and L<EVP_KEYEXCH_do_all_provided(3)>
+
+=item L<EVP_KEYMGMT_fetch(3)> and L<EVP_KEYMGMT_do_all_provided(3)>
+
+=item L<EVP_MAC_fetch(3)> and L<EVP_MAC_do_all_provided(3)>
+
+=item L<EVP_MD_fetch(3)> and L<EVP_MD_do_all_provided(3)>
+
+=item L<EVP_PKEY_CTX_new_from_pkey(3)>
+
+=item L<EVP_PKEY_Q_keygen(3)>
+
+=item L<EVP_Q_mac(3)> and L<EVP_Q_digest(3)>
+
+=item L<EVP_RAND(3)> and L<EVP_RAND_do_all_provided(3)>
+
+=item L<EVP_set_default_properties(3)>
+
+=item L<EVP_SIGNATURE_fetch(3)> and L<EVP_SIGNATURE_do_all_provided(3)>
+
+=item L<OSSL_CMP_CTX_new(3)> and L<OSSL_CMP_SRV_CTX_new(3)>
+
+=item L<OSSL_CRMF_ENCRYPTEDVALUE_get1_encCert(3)>
+
+=item L<OSSL_CRMF_MSG_create_popo(3)> and L<OSSL_CRMF_MSGS_verify_popo(3)>
+
+=item L<OSSL_CRMF_pbm_new(3)> and L<OSSL_CRMF_pbmp_new(3)>
+
+=item L<OSSL_DECODER_CTX_add_extra(3)> and L<OSSL_DECODER_CTX_new_for_pkey(3)>
+
+=item L<OSSL_DECODER_fetch(3)> and L<OSSL_DECODER_do_all_provided(3)>
+
+=item L<OSSL_ENCODER_CTX_add_extra(3)>
+
+=item L<OSSL_ENCODER_fetch(3)> and L<OSSL_ENCODER_do_all_provided(3)>
+
+=item L<OSSL_LIB_CTX_free(3)>, L<OSSL_LIB_CTX_load_config(3)> and L<OSSL_LIB_CTX_set0_default(3)>
+
+=item L<OSSL_PROVIDER_add_builtin(3)>, L<OSSL_PROVIDER_available(3)>,
+L<OSSL_PROVIDER_do_all(3)>, L<OSSL_PROVIDER_load(3)>,
+L<OSSL_PROVIDER_set_default_search_path(3)> and L<OSSL_PROVIDER_try_load(3)>
+
+=item L<OSSL_SELF_TEST_get_callback(3)> and L<OSSL_SELF_TEST_set_callback(3)>
+
+=item L<OSSL_STORE_attach(3)>
+
+=item L<OSSL_STORE_LOADER_fetch(3)> and L<OSSL_STORE_LOADER_do_all_provided(3)>
+
+=item L<RAND_get0_primary(3)>, L<RAND_get0_private(3)>, L<RAND_get0_public(3)>,
+L<RAND_set_DRBG_type(3)> and L<RAND_set_seed_source_type(3)>
+
+=back
+
+=head3 Providers
+
+Providers are described in detail here L<crypto(7)/Providers>.
+See also L<crypto(7)/OPENSSL PROVIDERS>.
+
+=head3 Fetching algorithms and property queries
+
+Implicit and Explicit Fetching is described in detail here
+L<crypto(7)/ALGORITHM FETCHING>.
+
+=head3 Deprecation of Low Level Functions
+
+A significant number of API's have been deprecated in OpenSSL 3.0.
```

Comment:
General usage seems to be APIs rather than API's - searching the github diff yields 60 hits for the former and 3 for the latter.